### ISSUE TITLE
Prediction using KNN 

### ISSUE THREAD

**Issue by freny24** [srimani-programmer/Breast-Cancer-Predictor#173]
Implementation of KNN algorithm for classification whether it is Malignant or Benign .
Dataset : https://www.kaggle.com/uciml/breast-cancer-wisconsin-data. (a different dataset can also be used!)

Comment down below if you are a GSSoc'22 participant and would like to contribute on this issue!

**Comment by Naidu-Saikiran** [comment]
I am a GSSoc'22 participant. I would like to contribute on this issue!

**Comment by Charu2510** [comment]
can i work this issue under GSSoC'22?

**Comment by freny24** [comment]
@Saikirannaidu225 describe how will you work on this issue?

**Comment by KareemElozeiri** [comment]
I am a GSSoc'22 participant, Can I work on this issue?

**Comment by KareemElozeiri** [comment]
I will process the data and feed it to the KNN algorithm that is already written in the sklearn library so that I do not re-invent that wheal and write the algorithm from scratch (approach description)

**Comment by prajwal-3-14159** [comment]
I would like to contribute, I am interested in participating in GSSoc 2021.

**Comment by Naidu-Saikiran** [comment]
> @Saikirannaidu225 describe how will you work on this issue?

I will use knn algorithm from sklearn library  to train the model. before doing it I will do the exploratory data analysis to understand the data and their relations .After that I will split the data into train and test data to train and test the KNN model.
I will use model metrics to evaluate the  performance of the algorithm

**Comment by inirah02** [comment]
I want to work on this issue
I'm GSSoC'22 Contributor

**Comment by mo1998** [comment]
> Implementation of KNN algorithm for classification whether it is Malignant or Benign . Dataset : https://www.kaggle.com/uciml/breast-cancer-wisconsin-data. (a different dataset can also be used!)
> 
> Comment down below if you are a GSSoc'22 participant and would like to contribute on this issue!

Hi, i am GSSoC'22 contributor and would like to work on this issue.
I have worked on similar project with same data set with accuracy score 96.5%.
Steps to be followed:
1- Analysis the data to get better understanding of parameters and thier relationships.
2- Drop unnecessary features.
3- Dealing with outliers.
4- Feature selection using VIF.
5- Train the model using KNN algorithm with (euclidean, manhattan, minkowski) metrics and (uniform,distance) weights. 
6- Evaluate model performance.

**Comment by nicolemd7** [comment]
Hi,I'm a GGSOC contributor I have made a pull request (#185) for this issue  using Neighborhood Component Analysis I have achieved 99% accuracy,please review

**Comment by nicolemd7** [comment]
I am a contributor can I be assigned this issue

**Comment by nicolemd7** [comment]
for this issue I plan on using Neighborhood Component Analysis for which the accuracy will go up to 99%

**Comment by dona-samir** [comment]
Hi, i am GSSoC'22 contributor and would like to work on this issue ,I have worked on knn model.
1. do data Analysis & preprocessing if need it .
2. check relation between features to drop unnecessary features.
3- Train the model using KNN algorithm with hyperparameter Tuning.
4- Evaluate model performance.

**Comment by dona-samir** [comment]
Hi, I am GSSoC'22 contributor ,I have worked on knn model.
1. do data Analysis & preprocessing if need it .
2. check relation between features to drop unnecessary features.
3- Train the model using KNN algorithm with hyperparameter Tuning.
4- Evaluate model performance.

**Comment by dona-samir** [comment]
Hi, I am GSSoC'22 contributor ,I have worked with knn model.
1. do data Analysis & preprocessing if need it .
2. check relation between features to drop unnecessary features.
3. Train the model using KNN algorithm with hyperparameter Tuning.
4.  Evaluate model performance.

**Comment by YatreeLadani** [comment]
Hello

I am a GSSoC22 contributor. I would like to work on this issue .I have already worked with KNN model in breast cancer project . Please do assign this to me.

1- Do data Analysis & preprocessing & scaling.
2- Check relation between features to drop unnecessary features.
3- Dealing with outliers.
4- Feature engineering.
5- Train the model using KNN algorithm with hyperparameter Tuning.
6- Evaluate model performance with matrix.

Thank You :)

**Comment by shrut-09** [comment]
hey, I would like to work on this issue. I am a GSSoC'22 contributor. I am a beginner and would like to work and learn

**Comment by miretteamin** [comment]
Hii!
I am a participant in GSSOC'22. Can you please assign me this issue.
I've implemented KNN before from scratch and used it in ZINDI competition of Blood Spectroscopy Classification and I've also used it digit recognizer.
Thank You.

**Comment by Neha-Niharika-Kar** [comment]
Hello. I am a GSSoC'22 contributor. I'm completely new to open source. I would like to work on this issue.

**Comment by freny24** [comment]
@Saikirannaidu225 issue has been assigned to you on first come first serve basis! do work on it

**Comment by navyasharma0203** [comment]
I would like to work on this issue. Can you assign this to me?

**Comment by ritikaaaa177** [comment]
Hello @freny24  I am a GSSOC'23 participant and as this issue is opened, i would like to work on it. Please provide me with the opportunity to work on it.
